From 52d441fac03cbcb63ff37c29e2794f9022ffd8bd Mon Sep 17 00:00:00 2001
From: Hasini Gunasinghe <hasinitg@google.com>
Date: Wed, 14 Oct 2020 14:49:58 +0000
Subject: [PATCH] [Q_asb_2021-01] Make mIsDeviceLockedForUser synchronized.

Bug: 169933423
Test: TBD/Treehugger passes
Change-Id: I8e1f57dd5ab8314801bdd62058c9ed5f761b7c55
(cherry picked from commit 8696c2ca1be068551a01008d0df642f5ccd86354)
---
 keystore/keystore_keymaster_enforcement.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/keystore/keystore_keymaster_enforcement.h b/keystore/keystore_keymaster_enforcement.h
index e7515a1f..b0dae48e 100644
--- a/keystore/keystore_keymaster_enforcement.h
+++ b/keystore/keystore_keymaster_enforcement.h
@@ -86,16 +86,19 @@ class KeystoreKeymasterEnforcement : public KeymasterEnforcement {
     }
 
     bool is_device_locked(int32_t userId) const override {
+        std::lock_guard<std::mutex> lock(is_device_locked_for_user_map_lock_);
         // If we haven't had a set call for this user yet, assume the device is locked.
         if (mIsDeviceLockedForUser.count(userId) == 0) return true;
         return mIsDeviceLockedForUser.find(userId)->second;
     }
 
     void set_device_locked(bool isLocked, int32_t userId) {
+        std::lock_guard<std::mutex> lock(is_device_locked_for_user_map_lock_);
         mIsDeviceLockedForUser[userId] = isLocked;
     }
 
   private:
+    mutable std::mutex is_device_locked_for_user_map_lock_;
     std::map<int32_t, bool> mIsDeviceLockedForUser;
 };
 
-- 
2.34.1

